packages/rum-core/src/domain/action/interactionSelectorCache.ts

```diff
@@ -0,0 +1,23 @@
+import { elapsed, ONE_SECOND, relativeNow } from '@datadog/browser-core'
+import type { RelativeTime } from '@datadog/browser-core'
+
+// Maximum duration for click actions
+export const CLICK_ACTION_MAX_DURATION = 10 * ONE_SECOND
+const cache = new Map<RelativeTime, string>()
+
+export const interactionSelectorCache = {
+  interactionSelectors: cache,
+  get: (relativeTimestamp: RelativeTime) => {
+    const selector = cache.get(relativeTimestamp)
+    cache.delete(relativeTimestamp)
+    return selector
+  },
+  set: (relativeTimestamp: RelativeTime, selector: string) => {
+    cache.set(relativeTimestamp, selector)
+    cache.forEach((_, relativeTimestamp) => {
+      if (elapsed(relativeTimestamp, relativeNow()) > CLICK_ACTION_MAX_DURATION) {
+        cache.delete(relativeTimestamp)
+      }
+    })
+  },
+}
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏
For stateful module, we tend to a export get and set function like in replayStats.ts. So it would look like:

```suggestion
export const CLICK_ACTION_MAX_DURATION = 10 * ONE_SECOND
const interactionSelectorCache = new Map<RelativeTime, string>()

export function getInteractionSelector(relativeTimestamp: RelativeTime) {
  const selector = interactionSelectorCache.get(relativeTimestamp)
  interactionSelectorCache.delete(relativeTimestamp)
  return selector
}

export function setInteractionSelector(relativeTimestamp: RelativeTime, selector: string) {
  interactionSelectorCache.set(relativeTimestamp, selector)
  clearOutdatedInteractionSelectors()
}

function clearOutdatedInteractionSelectors() {
  interactionSelectorCache.forEach((_, relativeTimestamp) => {
    if (elapsed(relativeTimestamp, relativeNow()) > CLICK_ACTION_MAX_DURATION) {
      interactionSelectorCache.delete(relativeTimestamp)
    }
  })
}

```

Reply:
`clearOutdatedInteractionSelectors` only used once and I think `set` is a bit misleading if we also clear. So I renamed the `set` function to be `updateInteractionSelector` and included both setting and clearing logic directly.